tests/test_pipelines.py

```diff
@@ -5,11 +5,21 @@
 import pytest
 from pytest_lazyfixture import lazy_fixture
 
-from pangeo_forge_recipes.executors.base import Pipeline, Stage
+from pangeo_forge_recipes.executors.base import Pipeline, PipelineExecutor, Stage
 from pangeo_forge_recipes.executors.dask import DaskPipelineExecutor
 from pangeo_forge_recipes.executors.function import FunctionPipelineExecutor
 from pangeo_forge_recipes.executors.prefect import PrefectPipelineExecutor
 
+# Only run Beam executor tests if Beam is in the current test environment.
+try:
+    from pangeo_forge_recipes.executors.beam import BeamPipelineExecutor
+except (ImportError, AttributeError):
+
+    class BeamPipelineExecutor(PipelineExecutor[None]):
+        @staticmethod
+        def compile(pipeline: Pipeline) -> None:
+            pytest.skip()
```

Comment:
I'm not wild about this way of getting to skip. It would be nice to have a solution that doesn't know anything about how to implement the class.

What if we used this pattern from pytest: https://docs.pytest.org/en/6.2.x/example/parametrize.html#indirect-parametrization-of-optional-implementations-imports ?

Redefine `Executor` as a parametrized fixture (rather than a parametrized argument). Something like

```python
@pytest.fixture(
    scope="session",
    params=[
        ("pangeo_forge_recipes.executors.dask", "DaskPipelineExecutor"),
        ("pangeo_forge_recipes.executors.function", "FunctionPipelineExecutor"),
        ("pangeo_forge_recipes.executors.prefect", "PrefectPipelineExecutor"),
        ("pangeo_forge_recipes.executors.beam", "BeamPipelineExecutor"),
    ]
)
def Executor(request):
    module = pytest.importorskip(request.param[0])
    return getattr(module, param[1])
```
(disclaimer: untested)

There may be an even simpler way to do it...

Reply:
Thanks for the suggestion. I'm not too familiar with `pytest`, and this worked with minor adjustments. I'm not sure if I know of a simpler method.